src/components/Modals/CreateItemModal/CreateItemModal.tsx

```diff
@@ -255,7 +256,16 @@ export default class CreateItemModal extends React.PureComponent<Props, State> {
   async updateThumbnail(category: WearableCategory) {
     const { model, contents } = this.state
     const url = URL.createObjectURL(contents![model!])
-    const { image: thumbnail } = await getModelData(url, { thumbnailType: getThumbnailType(category) })
+    const hasCustomThumbnail = contents && THUMBNAIL_PATH in contents
+    const isPngModel = model && model.endsWith('.png')
+
+    let thumbnail
+    if (contents && (hasCustomThumbnail || isPngModel)) {
+      thumbnail = await blobToDataURL(contents[THUMBNAIL_PATH] || contents[model!])
+    } else {
+      const { image } = await getModelData(url, { thumbnailType: getThumbnailType(category) })
+      thumbnail = image
+    }
```

Comment:
Maybe we can extract the following logic:
```ts
    const hasCustomThumbnail = contents && THUMBNAIL_PATH in contents
    const isPngModel = model && model.endsWith('.png')
```
into a function and reuse it in the three places we had to change 
- [248](https://github.com/decentraland/builder/pull/1297/files#diff-f14762c8fc5741ff060b0d666af2f4070db54123ca7ccfba537d984136c4834fR248)
- [260](https://github.com/decentraland/builder/pull/1297/files#diff-f14762c8fc5741ff060b0d666af2f4070db54123ca7ccfba537d984136c4834fR260)
- [364](https://github.com/decentraland/builder/pull/1297/files#diff-f14762c8fc5741ff060b0d666af2f4070db54123ca7ccfba537d984136c4834fR364)
?